src/v/config/configuration.cc

```diff
@@ -812,6 +812,24 @@ configuration::configuration()
       "Max https connection idle time (ms)",
       required::no,
       5s)
+  , cloud_storage_cache_directory(
```

Comment:
`cloud_storage_cache_directory` should have default value if it is not required

Reply:
std::nullopt maybe?

Reply:
done! also change type of cloud_storage_cache_directory from `data_directory_path` to `std::optional<ss::sstring>`